### PR TITLE
[MM-28303]: check if mmctl exists before checking on Github and reduce requests to Github

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

- Every `make` call would ping Github to find out the mmctl version to download. Each check made 4 requests to Github. So every `make` execution resulted in 4 requests to Github. This leads to frequent rate-limit errors from Github.
In this PR we check for the mmctl version only if mmctl doesn't already exist. We also print a more helpful error message.
- Reduce the number the number of requests to Github from 4 to 2. 

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-28303

### CODE REVIEW

**Comment by ashishbhate** [comment]
Works on my machine (Linux), please test on OSX (Darwin) and Windows.

I think I've tested all edge-cases, but more manual testing would be great.

**Comment by ashishbhate** [comment]
Looks like I've mucked up the make conditionals. I'll refactor and ping when ready.

**Comment by ashishbhate** [comment]
@jaydeland @agnivade  this is ready for review.

**Comment by jaydeland** [comment]
> @jaydeland @agnivade  this is ready for review. 

Thank you - will take a look.

File: Makefile [hunk 1]
```diff
@@ -188,24 +185,10 @@ prepackaged-plugins: ## Populate the prepackaged-plugins directory
...
-else ifeq ($(PLATFORM),Linux)
-	@echo Downloading prepackaged binary: https://github.com/mattermost/mmctl/releases/$(MMCTL_REL_TO_DOWNLOAD)
-	@MMCTL_FILE="linux_amd64.tar" && curl -f -O -L https://github.com/mattermost/mmctl/releases/download/$(MMCTL_REL_TO_DOWNLOAD)/$$MMCTL_FILE && tar -xvf $$MMCTL_FILE -C bin && rm $$MMCTL_FILE
-else ifeq ($(PLATFORM),Windows)
-	@echo Downloading prepackaged binary: https://github.com/mattermost/mmctl/releases/$(MMCTL_REL_TO_DOWNLOAD)
-	@MMCTL_FILE="windows_amd64.zip" && curl -f -O -L https://github.com/mattermost/mmctl/releases/download/$(MMCTL_REL_TO_DOWNLOAD)/$$MMCTL_FILE && unzip -o $$MMCTL_FILE -d bin && rm $$MMCTL_FILE
+	@echo "mmctl already exists in bin/mmctl not downloading a new version."
 else
-	@echo "mmctl error: can't detect OS"
+	@scripts/get_latest_release.sh 'mattermost/mmctl' 'release-' | xargs -0 scripts/download_mmctl_release.sh
```
Comment (agnivade) [inline Makefile:191]: I am a bit wary about piping the output directly to another script and doing the error checking there. Can we instead run `get_latest_release.sh`, and then check the exit code and run `download_mmctl_release.sh` only if the exit code is 0?

That should remove the brittle condition for checking for "error" in the input string.

**Review by agnivade — COMMENTED** [review]
As a sidenote, this is getting much too complicated to be expressed in Makefile logic. We need pure Go code using Magefiles for this which can be tested and readable.

File: Makefile (hunk 1 above)
Comment (ashishbhate) [reply Makefile]: @agnivade updated to use the previous command's exit code.

**Review by agnivade — APPROVED** [review]
Nice work

**Comment by ashishbhate** [comment]
@jaydeland gentle bump for review.

**Review by jaydeland — APPROVED** [review]
These are great enhancements - thank you!

**Comment by ashishbhate** [comment]
@lindalumitchell there isn't anything to QA here specifically. The PR only optimises some make targets for running the server in development.

**Review by lindalumitchell — APPROVED** [review]
